presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPruneCountAggregationOverScalar.java

```diff
@@ -0,0 +1,129 @@
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.assertions.BasePlanTest;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.iterative.rule.test.RuleTester;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.Assignments;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.facebook.presto.tpch.TpchColumnHandle;
+import com.facebook.presto.tpch.TpchTableHandle;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.FunctionKind.AGGREGATE;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCALE_FACTOR;
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
+
+public class TestPruneCountAggregationOverScalar extends BasePlanTest
+{
+    private RuleTester tester;
+    @BeforeClass
+    public void setUp()
+    {
+        tester = new RuleTester();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        closeAllRuntimeException(tester);
+        tester = null;
+    }
+    @Test
+    public void testDoesNotFireOnNonNestedAggregate()
+    {
+        tester.assertThat(new PruneCountAggregationOverScalar())
+                .on(p ->
+                        p.aggregation((a) ->
+                                a.source(p.tableScan(ImmutableList.of(), ImmutableMap.of()))
+                                .groupingSets(ImmutableList.of(ImmutableList.of()))
+                                .addAggregation(p.symbol("count_1", BigintType.BIGINT), new AggregationNode.Aggregation(
+                                        new FunctionCall(QualifiedName.of("count"), ImmutableList.of()),
+                                        new Signature("count", AGGREGATE, parseTypeSignature(StandardTypes.BIGINT)),
+                                        Optional.empty())))
+                ).doesNotFire();
+    }
+    @Test
+    public void testFiresOnNestedCountAggregate()
+    {
+        tester.assertThat(new PruneCountAggregationOverScalar())
+                .on(p ->
+                        p.aggregation((a) -> a.addAggregation(
+                                p.symbol("count_1", BigintType.BIGINT),
+                                new AggregationNode.Aggregation(
```

Comment:
[optional]

I do not like the fact you need to call constructors of `PlanNode` internals directly, it would be nice to have a better support for creating aggregation nodes in `PlanBuilder`

Can you please take a look and check if there is nothing there already which you could use? If there is nothing helpful there, maybe you could add something.

In current form this difficult to understand actual shape of plan as it is obscured with all the boilerplate like `ImmutableList.of()`.